admin/src/admin.js

```diff
@@ -74,13 +76,17 @@ class AdminUI extends Component {
   };
 
   state = {
-    showAutoEndSessionDialog: false
+    showAutoEndSessionDialog: false,
+    isAdmin: true
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     if (process.env.NODE_ENV !== "development" || qs.get("idle_timeout")) detectIdle();
     window.addEventListener("idle_detected", this.onIdleDetected);
     window.addEventListener("activity_detected", this.onActivityDetected);
+    const adminInfo = await getAdminInfo();
+    // Unauthorized account
+    if (adminInfo.status === 401) this.setState({ isAdmin: false });
```

Comment:
A little bit awkward to have `getAdminInfo` return the `Response` object in the error case (so that we can check the `status`) and the parsed json body in the success case. I don't have a better suggestion for folding in the status check/error handling but noting it because it stuck out to me.